pkg/controller/utils/component_test.go

```diff
@@ -1388,6 +1387,65 @@ var _ = Describe("Mocked client Component handler tests", func() {
 
 		Expect(mc.Index).To(Equal(4))
 	})
+
+	It("NetworkPolicy updates are omitted if there is no change", func() {
```

Comment:
PR lgtm, but it would be nice to have a similar test case for a scenario where there is a change.